From: Andreas Schwab <schwab@suse.de>
Date: Mon, 13 May 2013 15:31:45 +0200
Description: [PATCH] Add support for AAarch64
  Modified by Lisandro D. N. Pérez Meyer <lisandro@debian.org>
  The original patch also changed Source/WTF/wtf/dtoa/utils.h, but
  we are already doing this on
  double_conversion_correct_double_operations.patch
Origin: backport, https://bugs.webkit.org/show_bug.cgi?id=116030
Bug: https://bugs.webkit.org/show_bug.cgi?id=116030
        * wtf/Platform.h: Add support for AArch64.
Reviewed-by: Lisandro Damián Nicanor Pérez Meyer <lisandro@debian.org>
Last-Update: 2014-09-05

---
 Source/WTF/wtf/Platform.h |   11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

--- a/Source/WTF/wtf/Platform.h
+++ b/Source/WTF/wtf/Platform.h
@@ -324,6 +324,14 @@
 #define WTF_CPU_NEEDS_ALIGNED_ACCESS 1
 #endif
 
+/* CPU(AARCH64) - AArch64 */
+#if defined(__aarch64__)
+#define WTF_CPU_AARCH64 1
+#if defined(__AARCH64BE__)
+#define WTF_CPU_BIG_ENDIAN 1
+#endif
+#endif
+
 /* ==== OS() - underlying operating system; only to be used for mandated low-level services like 
    virtual memory, not to choose a GUI toolkit ==== */
 
@@ -882,7 +890,8 @@
     || CPU(ALPHA) \
     || CPU(SPARC64) \
     || CPU(S390X) \
-    || CPU(PPC64)
+    || CPU(PPC64) \
+    || CPU(AARCH64)
 #define WTF_USE_JSVALUE64 1
 #else
 #define WTF_USE_JSVALUE32_64 1
